django/db/backends/base/operations.py

```diff
@@ -30,6 +30,9 @@ class BaseDatabaseOperations:
         'SmallAutoField': (-32768, 32767),
         'AutoField': (-2147483648, 2147483647),
         'BigAutoField': (-9223372036854775808, 9223372036854775807),
+        'SmallSerialField': (-32768, 32767),
+        'SerialField': (-2147483648, 2147483647),
+        'BigSerialField': (-9223372036854775808, 9223372036854775807),
```

Comment:
Without these the test suite wouldn't run on other DB backends because the SerialModel would fail to validate. Is there some other way to prevent that from occurring, or could we use the `*IntegerField` internal types for the serial fields?

Reply:
`SerialField`\s shouldn't be tested on other backends.

Reply:
Agreed. How should I define a model for use in the postgres tests that either won't be seen or won't trip up model validators when running with a different backend?

Reply:
You can define it in `postgres_tests` and use `required_db_vendor`.

Reply:
I've already inherited from `postgres_tests.models.PostgreSQLModel` which sets `required_db_vendor`, but the problem persists. A Traceback from django-docker-box run with mariadb follows.
```
Traceback (most recent call last):
  File "tests/runtests.py", line 577, in <module>
    options.start_at, options.start_after, options.pdb, options.buffer,
  File "tests/runtests.py", line 315, in django_tests
    extra_tests=extra_tests,
  File "/tests/django/django/test/runner.py", line 698, in run_tests
    self.run_checks(databases)
  File "/tests/django/django/test/runner.py", line 636, in run_checks
    call_command('check', verbosity=self.verbosity, databases=databases)
  File "/tests/django/django/core/management/__init__.py", line 168, in call_command
    return command.execute(*args, **defaults)
  File "/tests/django/django/core/management/base.py", line 394, in execute
    output = self.handle(*args, **options)
  File "/tests/django/django/core/management/commands/check.py", line 69, in handle
    databases=options['databases'],
  File "/tests/django/django/core/management/base.py", line 419, in check
    databases=databases,
  File "/tests/django/django/core/checks/registry.py", line 70, in run_checks
    new_errors = check(app_configs=app_configs, databases=databases)
  File "/tests/django/django/core/checks/model_checks.py", line 34, in check_all_models
    errors.extend(model.check(**kwargs))
  File "/tests/django/django/db/models/base.py", line 1261, in check
    *cls._check_fields(**kwargs),
  File "/tests/django/django/db/models/base.py", line 1340, in _check_fields
    errors.extend(field.check(**kwargs))
  File "/tests/django/django/db/models/fields/__init__.py", line 1726, in check
    *super().check(**kwargs),
  File "/tests/django/django/db/models/fields/__init__.py", line 206, in check
    *self._check_validators(),
  File "/tests/django/django/db/models/fields/__init__.py", line 352, in _check_validators
    for i, validator in enumerate(self.validators):
  File "/tests/django/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/tests/django/django/db/models/fields/__init__.py", line 1748, in validators
    min_value, max_value = connection.ops.integer_field_range(internal_type)
  File "/tests/django/django/db/backends/base/operations.py", line 629, in integer_field_range
    return self.integer_field_ranges[internal_type]
KeyError: 'SmallSerialField'```

Reply:
This may also be a problem for users with multiple databases using different backends

Reply:
Unless there's some trick I'm not aware of, I think the cleanest two solutions would be to either use the IntegerField internal types, or to override the `validators` method serial fields inherit from `IntegerField`

Reply:
It's caused by inheriting from `IntegerField` & co. We can fix this with:
```diff
diff --git a/django/db/backends/base/operations.py b/django/db/backends/base/operations.py
index d9824735a7..3c7bef60cf 100644
@@ -616,7 +613,7 @@ class BaseDatabaseOperations:
         return a tuple of the (min_value, max_value) form representing the
         range of the column type bound to the field.
         """
-        return self.integer_field_ranges[internal_type]
+        return self.integer_field_ranges.get(internal_type, (None, None))
```

>  This is supposing that running validators on models with a non-matching required_db_vendor isn't a bug. 

Django doesn't try to run validators, it's only checking their implementation. This shouldn't be backend-specific. I don't think it's an issue.

Reply:
Ah, I see now. Thanks for your help and the suggested fix. I'll commit this here for now; is that okay or should I split it into a new PR later?